test/e2e/run-all.js

```diff
@@ -67,6 +67,14 @@ async function runningOnCircleCI(testPaths) {
   const changedOrNewTests = await filterE2eChangedFiles();
   console.log('Changed or new test list:', changedOrNewTests);
 
+  const hasOnlyMdOrCsvChanges = await checkOnlyMdOrCsvFiles();
+  if (hasOnlyMdOrCsvChanges) {
+    console.log(
+      'run-all.js info: Skipping test runs because run only has changes for MD or CSV files'
+    )
+    return {fullTestList: []};
```

Comment:
This addition is based on my interpretation on how test files are determined, not sure if this will work, thinking on way to test it

Reply:
Nits:
- It makes more sense to run this before `const changedOrNewTests = await filterE2eChangedFiles();`, because in certain paths you exit earlier and do less computation
- It would be nice to run `readChangedFiles()` only once and pass the result

Reply:
@HowardBraham thanks for suggestions! while I'm still working on testing this, I have included a boolean in return of the `filterE2eChangedFiles` promise, I think its not exactly the thing you suggested but if comparing file extensions is taking a lot time I can look into more optimizing it closer to your suggestion, open for any feedback 🙏